casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala

```diff
@@ -143,11 +144,11 @@ class RuntimeManager private (val emptyStateHash: ByteString, runtimeContainer:
       terms match {
         case deploy +: rem =>
           runtime.space.reset(hash)
-          val availablePhlos             = CostAccount(Integer.MAX_VALUE)
-          implicit val costAccountingAlg = CostAccountingAlg.unsafe[Task](availablePhlos) //FIXME this needs to come from the deploy params
-          val (phlosLeft, errors)        = injAttempt(deploy, runtime.reducer, runtime.errorLog)
-          val cost                       = phlosLeft.copy(cost = availablePhlos.cost.value - phlosLeft.cost)
-          val newCheckpoint              = runtime.space.createCheckpoint()
+          val availablePhlos = Cost(Integer.MAX_VALUE)
+          Await.ready(runtime.reducer.setAvailablePhlos(availablePhlos).runAsync, 1.second) // FIXME: This needs to come from the deploy params
```

Comment:
If you really have to do this (which scares me), why not `runSyncUnsafe`?

Reply:
You're right. I will, I have never thought about using that. 

I don't like this either but Casper is not abstracting over effect everywhere.

Reply:
My comment got lost, so (re)posting it:

I'm really wary of the await in here. Please explain what are you trying to achieve on a high level, and if and why mutating the internal state of reducer is not possible.

Reply:
Same would be for runSyncUnsafe ofc

Reply:
Well, we _are_ mutating an internal state of the reducer - only now it's `ChargingReducer` which also handles some logic regarding the `CostAccounting`.